websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,294 @@
+        .querySelector(
+          "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+        )
+        ?.getAttribute("src");
+
+    if (image) {
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = name;
+    }
+
+    presenceData.details = `Viewing movie ${name}`;
+    presenceData.largeImageKey = image ?? "logo";
+  } else if (pathname.startsWith("/manga")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
+    else if (paths[1] && paths[2]?.startsWith("vol-")) {
+      const tom = paths[2].replace("vol-", "");
+
+      if (paths[3].startsWith("chapter-")) {
+        const list = document.querySelectorAll(
+            "body > main.animated > ul#path > li > a > span"
+          ),
+          name = list[1].textContent,
+          page = document.querySelector(
+            "body > main.animated > div.wrapper > div.heading > b#num"
+          ).textContent;
+
+        presenceData.details = `Reading manga ${name}`;
+        presenceData.state = `Tom: ${tom}, Chapter: ${between(
+          list[3].textContent,
+          "Глава ",
+          " -"
+        )}, Page: ${page}`;
+        presenceData.buttons = [
+          {
+            label: "Read manga",
+            url: document.location.href
+          }
+        ];
+      } else {
+        const name = document.querySelector(
+          "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h2 > a.sub"
+        ).textContent;
+
+        if (name) presenceData.details = `Reading manga ${name}`;
+
+        presenceData.state = `Tom ${tom}`;
+        presenceData.buttons = [
+          {
+            label: "Read manga",
+            url: document.location.href
+          }
+        ];
+      }
+    } else if (hasNumber(paths[1])) {
+      const name = document.querySelector(
+          "body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading"
+        ).textContent,
+        image = document
+          .querySelector(
+            "body > main.animated > div.wrapper > article.rowView > aside.aside > div.cover-holder > img.abs"
+          )
+          ?.getAttribute("src");
+
+      if (name) presenceData.details = `Viewing manga ${name}`;
+
+      presenceData.largeImageKey = image ?? "logo";
+
+      if (image) {
+        presenceData.smallImageKey = "logo";
+        presenceData.smallImageText = name;
+      }
+
+      presenceData.buttons = [
+        {
+          label: "View manga",
+          url: document.location.href
+        }
+      ];
+    } else if (paths[1])
+      presenceData.details = `Viewing manga that starts with the keyboard ${paths[1].toUpperCase()}`;
+    else if (!paths[1]) presenceData.details = "Viewing manga";
+  }
+
+  if (presenceData.details) presence.setActivity(presenceData);
+  else presence.setActivity();
+});
+
+interface iframeData {
```

Comment:
Already requested this, follow interface naming ocnvention
```suggestion
interface IFrameData {
```